### ISSUE TITLE
chiller_absorption.py error due to null entry

### ISSUE THREAD

**Issue by Jack-Hawthorne** [architecture-building-systems/CityEnergyAnalyst#1825]
I am currently trying to run ``decentralized`` from cli however it seems that a null entry in one of the reference case input databases is throwing off the script. There are also some divide by zero errors than come up also. See below:

```
(cea) C:\Users\Jack>cea trace-inputlocator --scripts decentralized
City Energy Analyst version 2.9.2
Running `cea trace-inputlocator` with the following parameters:
- general:scenario = C:\reference-case-open\baseline
  (default: C:\reference-case-open\baseline)
- trace-inputlocator:scripts = ['decentralized']
  (default: ['data-helper', 'demand', 'emissions'])
- trace-inputlocator:graphviz-output-file = C:\reference-case-open\baseline/outputs/trace_inputlocator.output.gv
  (default: C:\reference-case-open\baseline/outputs/trace_inputlocator.output.gv)
- trace-inputlocator:meta-output-file = C:\reference-case-open\baseline/outputs/trace_inputlocator.output.yml
  (default: C:\reference-case-open\baseline/outputs/trace_inputlocator.output.yml)
City Energy Analyst version 2.9.2
Running `cea decentralized` with the following parameters:
- general:scenario = C:\reference-case-open\baseline
  (default: C:\reference-case-open\baseline)
- general:region = CH
  (default: CH)
- general:weather = c:\users\jack\documents\github\cityenergyanalyst\cea\databases\weather\Zug.epw
  (default: c:\users\jack\documents\github\cityenergyanalyst\cea\databases\weather\Zug.epw)
- decentralized:ahuflag = False
  (default: False)
- decentralized:aruflag = False
  (default: False)
- decentralized:scuflag = False
  (default: False)
- decentralized:ahuaruflag = False
  (default: False)
- decentralized:ahuscuflag = False
  (default: False)
- decentralized:aruscuflag = False
  (default: False)
Running decentralized model for buildings with scenario = C:\reference-case-open\baseline
C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)
B01
3.3671602767 seconds process time for the Substation Routine.
B01
3.29787225872 seconds process time for the Substation Routine.
B01
2.8959399732 seconds process time for the Substation Routine.
B01
3.47161368967 seconds process time for the Substation Routine.
B01
3.4970826704 seconds process time for the Substation Routine.
B01
3.65972566222 seconds process time for the Substation Routine.
B01
3.58344799985 seconds process time for the Substation Routine.
B01  decentralized building simulation with configuration: AHU + ARU + SCU
decentralized building simulation with configuration: AHU + ARU + SCU cost calculations
B02
4.16223296568 seconds process time for the Substation Routine.
B02
4.03686949947 seconds process time for the Substation Routine.
B02
3.33284650729 seconds process time for the Substation Routine.
B02
3.63618947901 seconds process time for the Substation Routine.
B02
3.55790093922 seconds process time for the Substation Routine.
B02
3.5758307296 seconds process time for the Substation Routine.
B02
3.72642381684 seconds process time for the Substation Routine.
B02  decentralized building simulation with configuration: AHU + ARU + SCU
decentralized building simulation with configuration: AHU + ARU + SCU cost calculations
B03
3.62134533801 seconds process time for the Substation Routine.
B03
3.56177976092 seconds process time for the Substation Routine.
B03
2.96632622977 seconds process time for the Substation Routine.
B03
3.72139683661 seconds process time for the Substation Routine.
B03
3.55719224679 seconds process time for the Substation Routine.
B03
3.53834981758 seconds process time for the Substation Routine.
B03
3.76394568882 seconds process time for the Substation Routine.
B03  decentralized building simulation with configuration: AHU + ARU + SCU
decentralized building simulation with configuration: AHU + ARU + SCU cost calculations
B06
7.68752125331 seconds process time for the Substation Routine.
B06
6.09669960956 seconds process time for the Substation Routine.
B06
4.22402659325 seconds process time for the Substation Routine.
B06
3.55576334196 seconds process time for the Substation Routine.
B06
6.7086975462 seconds process time for the Substation Routine.
B06
6.78068209406 seconds process time for the Substation Routine.
B06
4.95316613995 seconds process time for the Substation Routine.
B06  decentralized building simulation with configuration: AHU + ARU + SCU
decentralized building simulation with configuration: AHU + ARU + SCU cost calculations
B04
3.27352199656 seconds process time for the Substation Routine.
B04
3.29193477299 seconds process time for the Substation Routine.
B04
2.91522741517 seconds process time for the Substation Routine.
B04
3.43244488647 seconds process time for the Substation Routine.
B04
3.43340019191 seconds process time for the Substation Routine.
B04
3.44212764741 seconds process time for the Substation Routine.
B04
3.58516618432 seconds process time for the Substation Routine.
B04  decentralized building simulation with configuration: AHU + ARU + SCU
decentralized building simulation with configuration: AHU + ARU + SCU cost calculations
B05
5.35472679928 seconds process time for the Substation Routine.
B05
5.23161911686 seconds process time for the Substation Routine.
B05
5.44725532085 seconds process time for the Substation Routine.
B05
4.98525953887 seconds process time for the Substation Routine.
B05
5.72422563299 seconds process time for the Substation Routine.
B05
6.02161277234 seconds process time for the Substation Routine.
B05
6.37076005876 seconds process time for the Substation Routine.
B05  decentralized building simulation with configuration: AHU + ARU + SCU
c:\users\jack\documents\github\cityenergyanalyst\cea\technologies\cooling_tower.py:51: RuntimeWarning: invalid value encountered in double_scalars
  qpartload = qhotdot_W / Qdesign_W
c:\users\jack\documents\github\cityenergyanalyst\cea\technologies\cooling_tower.py:55: RuntimeWarning: invalid value encountered in double_scalars
  qpartload + 0.0026
decentralized building simulation with configuration: AHU + ARU + SCU cost calculations
B07
c:\users\jack\documents\github\cityenergyanalyst\cea\technologies\substation.py:516: RuntimeWarning: divide by zero encountered in double_scalars
  tco_0 = Qnom / cc_0 + tci_0
C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\numpy\lib\scimath.py:262: RuntimeWarning: invalid value encountered in log
  return nx.log(x)
c:\users\jack\documents\github\cityenergyanalyst\cea\technologies\substation.py:766: RuntimeWarning: invalid value encountered in true_divide
  dTm = (dT1 - dT2) / scipy.log(dT1 / dT2)
C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\numpy\lib\function_base.py:2048: RuntimeWarning: invalid value encountered in calc_HEX_cooling (vectorized)
  outputs = ufunc(*inputs)
C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\numpy\lib\function_base.py:2048: RuntimeWarning: invalid value encountered in calc_DC_HEX_mix (vectorized)
  outputs = ufunc(*inputs)
4.72198478253 seconds process time for the Substation Routine.
B07
4.71625465653 seconds process time for the Substation Routine.
B07
4.89685815869 seconds process time for the Substation Routine.
B07
5.51487806096 seconds process time for the Substation Routine.
B07
5.27832028375 seconds process time for the Substation Routine.
B07
5.07814091331 seconds process time for the Substation Routine.
B07
5.59646600986 seconds process time for the Substation Routine.
C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\numpy\lib\function_base.py:2048: RuntimeWarning: invalid value encountered in calc_new_load (vectorized)
  outputs = ufunc(*inputs)
B07  decentralized building simulation with configuration: AHU + ARU + SCU
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\Scripts\cea-script.py", line 11, in <module>
    load_entry_point('cityenergyanalyst', 'console_scripts', 'cea')()
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\interfaces\cli\cli.py", line 65, in main
    script_module.main(config)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\tests\trace_inputlocator.py", line 52, in main
    script_func(config)  # <------------------------------ this is where we run the script!
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\api.py", line 26, in script_runner
    script_module.main(config)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\optimization\preprocessing\decentralized_building_main.py", line 53, in main
    disconnected_building_main(locator=locator, building_names=building_names, config=config, prices=prices, lca=lca)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\optimization\preprocessing\decentralized_building_main.py", line 37, in disconnected_building_main
    decentralized_buildings_cooling.disconnected_buildings_cooling_main(locator, building_names, config, prices, lca)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\optimization\preprocessing\decentralized_buildings_cooling.py", line 946, in disconnected_buildings_cooling_main
    locator, config)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\technologies\chiller_absorption.py", line 88, in calc_chiller_main
    number_of_chillers = int(ceil(input_conditions['q_chw_W'] / max_chiller_size))
ValueError: cannot convert float NaN to integer
```

**Comment by shanshanhsieh** [comment]
hi Jack,
are you running the reference-case-open with `has-cooling-season=TRUE` in `baseline/database/CH/archetypes/system_controls.xlsx`?

**Comment by Jack-Hawthorne** [comment]
@shanshanhsieh i am indeed running with ``has-cooling-season=TRUE`` which is the default it appears. The error only occurs while iterating over B07. I don't know if that's any help.

**Comment by shanshanhsieh** [comment]
@Jack-Hawthorne could you please share the `Total_demand.xlsx` file?

**Comment by Jack-Hawthorne** [comment]
@shanshanhsieh sure thing, i'll just link here as text
[Total_demand.txt](https://github.com/architecture-building-systems/CityEnergyAnalyst/files/2970410/Total_demand.txt)

**Comment by Jack-Hawthorne** [comment]
i've also tried running this with the following config and am getting the following:
```
(cea) C:\Users\Jack\Documents\GitHub\CityEnergyAnalyst\docs>cea decentralized
City Energy Analyst version 2.9.2a
Running `cea decentralized` with the following parameters:
- general:scenario = c:\reference-case-open\WTP_CBD_h
  (default: C:\reference-case-open\baseline)
- general:region = SG
  (default: CH)
- general:weather = c:\users\jack\documents\github\cityenergyanalyst\cea\databases\weather\Singapore.epw
  (default: c:\users\jack\documents\github\cityenergyanalyst\cea\databases\weather\Zug.epw)
- decentralized:ahuflag = True
  (default: False)
- decentralized:aruflag = True
  (default: False)
- decentralized:scuflag = True
  (default: False)
- decentralized:ahuaruflag = True
  (default: False)
- decentralized:ahuscuflag = True
  (default: False)
- decentralized:aruscuflag = True
  (default: False)
Running decentralized model for buildings with scenario = c:\reference-case-open\WTP_CBD_h
C:\Users\Jack\Miniconda2\envs\cea\lib\site-packages\pandas\io\excel.py:329: FutureWarning: The `sheetname` keyword is deprecated, use `sheet_name` instead
  **kwds)
Traceback (most recent call last):
  File "C:\Users\Jack\Miniconda2\envs\cea\Scripts\cea-script.py", line 11, in <module>
    load_entry_point('cityenergyanalyst', 'console_scripts', 'cea')()
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\interfaces\cli\cli.py", line 65, in main
    script_module.main(config)
  File "c:\users\jack\documents\github\cityenergyanalyst\cea\optimization\preprocessing\decentralized_building_main.py", line 51, in main
    lca = LcaCalculations(locator, config)
TypeError: __init__() takes exactly 4 arguments (3 given)
```
not sure if someone has changed the LcaCalculations recently to require 3 args but this is another issue

**Comment by JIMENOFONSECA** [comment]
@shanshanhsieh any ideas? @daren-thomas I believe you modified the name to LcaCalculations in one of the latest PR's. is that so?

**Comment by daren-thomas** [comment]
@Jack-Hawthorne @JIMENOFONSECA @shanshanhsieh my bad. I'm super sorry - I refactored something and missed three invocations of LcaCalculations.

I've fixed this in #1843
